// react-native-image-filter-kit/dist/color-matrix-filters/shapes.js
import { scalar, color, scalarVector } from '../common/inputs';
import { Common } from '../common/shapes';
const WithAmount = Object.assign({ amount: scalar }, Common);
export const shapes = {
    ColorMatrix: Object.assign({ matrix: scalarVector }, Common),
    Normal: Common,
    RGBA: Object.assign({ red: scalar, green: scalar, blue: scalar, alpha: scalar }, Common),
    Saturate: WithAmount,
    HueRotate: WithAmount,
    LuminanceToAlpha: Common,
    Invert: Common,
    Grayscale: WithAmount,
    Sepia: WithAmount,
    Nightvision: Common,
    Warm: Common,
    Cool: Common,
    Brightness: WithAmount,
    Contrast: WithAmount,
    Temperature: WithAmount,
    Tint: WithAmount,
    Threshold: WithAmount,
    Technicolor: Common,
    Polaroid: Common,
    ToBGR: Common,
    Kodachrome: Common,
    Browni: Common,
    Vintage: Common,
    Night: WithAmount,
    Predator: WithAmount,
    Lsd: Common,
    ColorTone: Object.assign({ desaturation: scalar, toned: scalar, lightColor: color, darkColor: color }, Common),
    DuoTone: Object.assign({ firstColor: color, secondColor: color }, Common),
    Protanomaly: Common,
    Deuteranomaly: Common,
    Tritanomaly: Common,
    Protanopia: Common,
    Deuteranopia: Common,
    Tritanopia: Common,
    Achromatopsia: Common,
    Achromatomaly: Common
};
